uploadNeuroDB/NeuroDB/MRIProcessingUtility.pm

```diff
@@ -1997,6 +1997,60 @@ sub isValidMRIProtocol  {
     }
 }
 
+=pod
+
+=head3 is_file_unique($file, $upload_id)
+
+Queries the C<files> and C<parameter_file> tables to make sure that no imaging
+datasets with the same C<SeriesUID> and C<EchoTime> or the same C<MD5sum> hash
+can be found in the database already. If there is a match, it will return a
+message with the information about why the file is not unique. If there is no
+match, then it will return undef.
+
+INPUTS:
+  - $file     : the file object from the C<NeuroDB::File> package
+  - $upload_id: the C<UploadID> associated to the file
+
+RETURNS: a message with the reason why the file is not unique or undef
+
+=cut
+
+sub is_file_unique {
+
+    my $this = shift;
+    my ($file, $upload_id) = @_;
+
+    my $seriesUID = $file->getParameter( 'series_instance_uid' );
+    my $echo_time = $file->getParameter( 'echo_time'           );
+
+    # check that no files are already in the files table with the same SeriesUID
+    # and EchoTime
+    my $query     = "SELECT File FROM files WHERE SeriesUID=? AND EchoTime=?";
+    my $sth       = ${$this->{'dbhr'}}->prepare( $query );
```

Comment:
Not really related to your PR but it is very annoying to have to dereference `$this->{'dbhr'}` every time we use it (as in `${ $this->{'dbhr'} }`). The constructor for the MRIProcessingUtility class should receive the `$dbhr` argument and not a reference to it and the usage of this variable would be simplified greatly. I opened [Redmine ticket #15738](https://redmine.cbrain.mcgill.ca/issues/15738) for that.